### ISSUE TITLE
Changing from IAM Certificate to ACM Certificate

### ISSUE THREAD

**Issue by himeshladva-ni** [kubernetes-sigs/aws-alb-ingress-controller#364]
Trying to change the `alb.ingress.kubernetes.io/certificate-arn` from an ARN referencing an IAM certificate to one referencing an ACM certificate and get the following error:

```
I0413 15:35:49.092745       1 listener.go:104] [ALB-INGRESS] [dev/alb-ingress] [INFO]: Start Listener modification.
E0413 15:35:49.092846       1 listener.go:174] [ALB-INGRESS] [dev/alb-ingress] [ERROR]: Failed Listener modification: InvalidParameter: 1 validation error(s) found.
E0413 15:35:49.092857       1 listener.go:174] [ALB-INGRESS] [dev/alb-ingress] [ERROR]: - missing required field, ModifyListenerInput.DefaultActions[0].TargetGroupArn.
E0413 15:35:49.092863       1 listener.go:174] [ALB-INGRESS] [dev/alb-ingress] [ERROR]: .
panic: runtime error: index out of range
 goroutine 186 [running]:
github.com/coreos/alb-ingress-controller/pkg/alb/listener.(*Listener).modify(0xc420789500, 0xc420293d48, 0x1c, 0x0)
	/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/pkg/alb/listener/listener.go:176 +0x1de
github.com/coreos/alb-ingress-controller/pkg/alb/listener.(*Listener).Reconcile(0xc420789500, 0xc420293d48, 0xc420052600, 0x66eab3)
	/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/pkg/alb/listener/listener.go:105 +0xce
github.com/coreos/alb-ingress-controller/pkg/alb/listeners.Listeners.Reconcile(0xc42000ca98, 0x1, 0x1, 0xc420293ed8, 0xc42063c0a0, 0x4, 0x4, 0x0, 0x0)
	/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/pkg/alb/listeners/listeners.go:44 +0x133
github.com/coreos/alb-ingress-controller/pkg/alb/loadbalancer.(*LoadBalancer).Reconcile(0xc420175500, 0xc420293f60, 0xc4201a97b0, 0x7ff1116ba4b0, 0x0)
	/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/pkg/alb/loadbalancer/loadbalancer.go:213 +0x29b
github.com/coreos/alb-ingress-controller/pkg/albingress.(*ALBIngress).Reconcile(0xc4201bbc30, 0xc420224fb8)
	/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/pkg/albingress/albingress.go:318 +0xb4
github.com/coreos/alb-ingress-controller/pkg/controller.(*ALBController).update.func1(0xc420651cd0, 0xc4201bbc30)
	/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/pkg/controller/alb-controller.go:175 +0xa9
created by github.com/coreos/alb-ingress-controller/pkg/controller.(*ALBController).update
	/home/travis/gopath/src/github.com/coreos/alb-ingress-controller/pkg/controller/alb-controller.go:176 +0x4cd
```

To solve, you have to completely remove the Ingress resource and create it from scratch.

**Comment by chrishiestand** [comment]
I had the same problem when going from no TLS cert to adding `alb.ingress.kubernetes.io/certificate-arn`. I had to delete and recreate the ingress.

**Comment by bigkraig** [comment]
Can you see if this was fixed in `3f87a2d`? I am planning to cut another beta from this version if no new bugs are detected at Ticketmaster. I've tested an ingress by adding an HTTPS port and `certificate-arn` to an existing ingress, then removing the arn, and it has worked as it should.

**Comment by bigkraig** [comment]
```I0627 14:04:57.295314   49979 alb-controller.go:186] [ALB-INGRESS] [sync] [DEBUG]: Requesting Lock.
I0627 14:04:57.295328   49979 alb-controller.go:186] [ALB-INGRESS] [sync] [DEBUG]: Lock was available.
I0627 14:04:59.283040   49979 alb-controller.go:208] [ALB-INGRESS] [controller] [DEBUG]: Ingress count: 3
I0627 14:04:59.283475   49979 listener.go:82] [ALB-INGRESS] [prd354/tm-aws-limits] [INFO]: Start Listener creation.
I0627 14:04:59.693189   49979 listener.go:87] [ALB-INGRESS] [prd354/tm-aws-limits] [INFO]: Completed Listener creation. ARN: arn:aws:elasticloadbalancing:us-east-1:XXX:listener/app/sandbox3-prd354-tmawslimit-e807/28f22d937415cf1f/706ae72fc75378c7 | Port: 443 | Proto: HTTPS.
I0627 14:04:59.693232   49979 rule.go:110] [ALB-INGRESS] [prd354/tm-aws-limits] [INFO]: Start Rule creation.
I0627 14:04:59.693241   49979 event.go:218] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"prd354", Name:"tm-aws-limits", UID:"5d3ccbe3-41d9-11e8-99ae-0e3a92b39dc4", APIVersion:"extensions", ResourceVersion:"91558106", FieldPath:""}): type: 'Normal' reason: 'CREATE' 443 listener created
I0627 14:04:59.828399   49979 rule.go:115] [ALB-INGRESS] [prd354/tm-aws-limits] [INFO]: Completed Rule creation. Rule Priority: "1" | Condition: [{    Field: "path-pattern",    Values: ["/"]  }]
I0627 14:04:59.828425   49979 alb-controller.go:228] [ALB-INGRESS] [sync] [DEBUG]: Unlock'd.
```

```
I0627 14:06:09.841421   49979 alb-controller.go:186] [ALB-INGRESS] [sync] [DEBUG]: Requesting Lock.
I0627 14:06:09.841428   49979 alb-controller.go:186] [ALB-INGRESS] [sync] [DEBUG]: Lock was available.
I0627 14:06:10.488916   49979 alb-controller.go:208] [ALB-INGRESS] [controller] [DEBUG]: Ingress count: 3
I0627 14:06:10.489295   49979 types.go:35] [ALB-INGRESS] [util] [DEBUG]: DeepEqual("HTTPS", "HTTP") found inequality
I0627 14:06:10.489329   49979 listener.go:209] [ALB-INGRESS] [prd354/tm-aws-limits] [DEBUG]: Protocol needs to be changed ("HTTPS" != "HTTP")
I0627 14:06:10.489343   49979 listener.go:92] [ALB-INGRESS] [prd354/tm-aws-limits] [INFO]: Start Listener modification.
I0627 14:06:10.619846   49979 listener.go:97] [ALB-INGRESS] [prd354/tm-aws-limits] [INFO]: Completed Listener modification. ARN: arn:aws:elasticloadbalancing:us-east-1:XXX:listener/app/sandbox3-prd354-tmawslimit-e807/28f22d937415cf1f/706ae72fc75378c7 | Port: 443 | Proto: HTTP.
I0627 14:06:10.619899   49979 alb-controller.go:228] [ALB-INGRESS] [sync] [DEBUG]: Unlock'd.
```
